nsxt/nsxt_vcenter_moids.py

```diff
@@ -0,0 +1,138 @@
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions
+# of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+# TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+# CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+
+__author__ = 'VJ49'
+
+from pyVmomi import vim, vmodl
+from pyVim import connect
+from pyVim.connect import SmartConnect, SmartConnectNoSSL
+
+import json, time
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, request
+from ansible.module_utils._text import to_native
+
+from ansible.module_utils.basic import *
+
+import logging
+logger = logging.getLogger('Vcenter Moids')
+hdlr = logging.FileHandler('/var/log/chaperone/ChaperoneNSXtLog.log')
+formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+hdlr.setFormatter(formatter)
+logger.addHandler(hdlr) 
+logger.setLevel(10)
+
+""" 
+    VIM_TYPES = {'datacenter': [vim.Datacenter],
+             'dvs_name': [vim.dvs.VmwareDistributedVirtualSwitch],
+             'datastore_name': [vim.Datastore],
+             'resourcepool_name': [vim.ResourcePool],
+             'portgroup_name': [vim.dvs.DistributedVirtualPortgroup, vim.Network]} """
+
+
+def get_all_objs(module,content, vimtype):
+    obj = {}
+    container = content.viewManager.CreateContainerView(content.rootFolder,
+                                                         vimtype, True)
+    for managed_object_ref in container.view:
+        obj.update({managed_object_ref: managed_object_ref.name})
+    return obj
+
+
+def find_object_by_name(module,content, object_name):
+    try:
+    	if(object_name == module.params['datacenter']):
+    	    vmware_objects = get_all_objs(module,content,[vim.Datacenter])
+    	elif (object_name == module.params['cluster']):
+    	    vmware_objects = get_all_objs(module,content,[vim.ComputeResource])
+    	elif (object_name == module.params['datastore']):
+            vmware_objects = get_all_objs(module,content,[vim.Datastore])
+    	elif (object_name == module.params['portgroup1'] or module.params['portgroup2'] or module.params['portgroup3'] ):
+            vmware_objects = get_all_objs(module,content,[vim.dvs.DistributedVirtualPortgroup, vim.Network])
+ 
+    	for object in vmware_objects:
+            if object.name == object_name:
+            	logger.info('object: %s',object.name)
+            	return object
+    	return None
```

Comment:
Could this module be replaced by https://docs.ansible.com/ansible/latest/modules/vmware_guest_find_module.html or one of the `_facts` modules from https://github.com/vmware/ansible-for-nsxt?  I'm not sure if that's possible or not, but would be good if we can avoid duplication.

Reply:
Item added to the backlog for future release.